### ISSUE TITLE
CreateMany dont respect the unique fields

### ISSUE THREAD

**Issue by HastaCs** [strapi/strapi#15181]
### Required System information

- Node.js version: 16.12
- NPM version: 8.1
- Strapi version: 4.5.4
- Database: sqlite3
- Operating system: Windows 10

### Describe the bug

The createMany operation from Bulk Operations are not respecting the fields marked as unique.
[Bulk Operations documentation](https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/query-engine/bulk-operations.html#createmany)

### Expected behavior

An error or not insert this rows, or a new field in response called 'failRows' or something like that

### Screenshots
![imagen](https://user-images.githubusercontent.com/53431308/207708428-9a7c22b0-93aa-4c1d-82a5-2fe01944ffe1.png)
![imagen](https://user-images.githubusercontent.com/53431308/207708262-7eb218ff-0399-4454-8d26-e6d3d34571a9.png)

### Code snippets
```

coinsAction: async (ctx, next) => {
    const data = await axios.get("https://api.coincap.io/v2/assets?limit=500&offset=0");
    const coinsToInsert = data.data.data.map((coin) => ({
      name: coin.name,
      code: coin.symbol,

    }));

    const inserted = await strapi.db.query("api::coin.coin").createMany({
      data: coinsToInsert,
    });

    ctx.body = `Data inserted ${inserted.count}`;
  },
```

### Additional context

I have created a basic strapi project with the api endpoints and models for test it.
[GITHUB](https://github.com/HastaCs/strapi-createmany-bug)

**Comment by bolg55** [comment]
please provide explicit steps to reproduce.

**Comment by HastaCs** [comment]
> please provide explicit steps to reproduce.

Sure sorry. I give you a manual way and a repository way.

**_MANUAL_**
1. Create a clean installacion of strapi :**npx create-strapi-app@latest strapi-bug --quickstart**
2. Create a new collection type called people with 2 fields:  **name**:string and **age**:integer(required and unique)
3.  Modify the **src/api/people/controllers/people.js** with this
```
'use strict';

const { createCoreController } = require('@strapi/strapi').factories;

module.exports = createCoreController('api::people.people', ({ strapi }) => ({

  async find(ctx) {
    const people = [{ name: "jhon", age: 20 }, { name: "peter", age: 20 }]

    await strapi.db.query("api::people.people").createMany({
      data: people,
    });
    ctx.body = `people inserted`;
  },
}))
```
4.Make this route public from strapi admin in  settings/roles/public/   People and click in find.
5. In a brower call the endpoint **http://localhost:1337/api/peoples** and there will be people with the same age

**_REPOSITORY_**
Or you can download this repo with all the previous changes:

1. Clone repository [Github](https://github.com/HastaCs/strapi-bug.git)
2. npm install
3. npm run develop
4. admin user is test@test.com and password is A123456a
5. call http://localhost:1337/api/peoples many times.

Greetings

**Comment by HastaCs** [comment]
ok I saw that the problem is not CreateMany, the problem is that when I mark as UNIQUE on a field, this does not create a UNIQUE index in the database. regards, at least in SQLITE3

**Comment by Marc-Roig** [comment]
@HastaCs Exactly, the uniqueness is validated inside the application, for reasons like:

- Different locales are stored in the same table, and unique fields have to be unique accross the same locale not all of them.

Because of that , those validations are done in the entityService, and so by using any query Engine method you can override the unique validations.
